modules/gateway/rpc.go

```diff
@@ -141,28 +154,27 @@ func (g *Gateway) threadedListenPeer(p *peer) {
 		}
 	}()
 
-	if g.threads.Add() != nil {
-		return
-	}
-	defer g.threads.Done()
-
 	for {
 		conn, err := p.accept()
 		if err != nil {
-			g.log.Println("WARN: lost connection to peer", p.NetAddress)
-			return
+			g.log.Debugln("Peer connection closed:", p.NetAddress)
```

Comment:
It's a common thing for peers to disconnect, not really something that you need to be paying attention to unless you are losing most of your peers.